### ISSUE TITLE
Error with build `:host-context` with spaced css selector

### ISSUE THREAD

**Issue by Dok11** [angular/angular#37078]
# 🐞 bug report

## 🔬 Minimal Reproduction
css code:
```css
:host-context(.aaa .bbb) {
  .my-class { color: red; }
}
```

Build into rule without spaces:
```
.aaa.bbb[_nghost-%COMP%] .my-class[_ngcontent-%COMP%],
.aaa.bbb [_nghost-%COMP%] .my-class[_ngcontent-%COMP%]{ color: red }
```

I think host context should save own css selector without modification.
In serve mode it works correctly.

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 8.2.2
Node: 10.16.3
OS: win32 x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.801.3
@angular-devkit/build-angular     0.801.3
@angular-devkit/build-optimizer   0.801.3
@angular-devkit/build-webpack     0.801.3
@angular-devkit/core              8.1.3
@angular-devkit/schematics        8.2.2
@angular/cdk                      8.2.3
@angular/cli                      8.2.2
@ngtools/webpack                  8.1.3
@schematics/angular               8.2.2
@schematics/update                0.802.2
rxjs                              6.5.3
typescript                        3.5.3
webpack                           4.35.2
</code></pre>

**Comment by pkozlowski-opensource** [comment]
@Dok11 could you please share a reproduction scenario (minimal GitHub repro or, better yet - stackblitz if reproductible) with the lates Angular 9.x/10 (a lot changed from 8.x)?

I was trying to reproduce the problem in question using stackblitz (see: https://stackblitz.com/edit/angular-ivy-xun6pq) and couldn't. With the following CSS in the component's styles:

```css
    :host-context(.aaa .bbb) {
      .my-class { color: red; }
    }
```

I'm seeing the following `<style>` tag generated using stackblitz linked above:

```html
<style>.aaa[_nghost-hlu-c58]   .bbb[_ngcontent-hlu-c58], .aaa   .bbb   [_nghost-hlu-c58] {
      .my-class { color: red; }
    }</style>
```

In short: try the latest version of Angular and if you still face problems, please share a minimal reproduce scenario. For now this issue is not actionable as-is.

**Comment by Dok11** [comment]
```html
<style>.aaa[_nghost-hlu-c58]   .bbb[_ngcontent-hlu-c58], .aaa   .bbb   [_nghost-hlu-c58] {
      .my-class { color: red; }
    }</style>
```

It's not correct style. Maybe it should be like this:
```css
.aaa[_nghost-hlu-c58] .bbb[_ngcontent-hlu-c58] .my-class { color: red; }
.aaa .bbb [_nghost-hlu-c58] .my-class { color: red; }
```

> I was trying to reproduce the problem in question using stackblitz.

Stackblitz can make build? In serve mode all works fine.

**Comment by pkozlowski-opensource** [comment]
> Stackblitz can make build? In serve mode all works fine.

Ut is definitively capable of doing AoT compilation of Angular components, if this is what you mean by "build". Again, at this point it is not clear where the root of the problem is - it might be as well in a CSS processing step of your build if you happen to have one.

In any case we need a reproduce scenario (stackblitz or a minimal GitHub repo) - we can't fix bugs that we can't reproduce and understand...

**Comment by pkozlowski-opensource** [comment]
I'm going to close this one for now as we haven't heard back with a minimal reproduce scenario. Happy to re-open if / when we've got one!
